aws/adapter.go

```diff
@@ -725,3 +727,36 @@ func generateDefaultFilters(clusterId string) []*ec2.Filter {
 		},
 	}
 }
+
+// We look to the same string used for instance filtering, however, we are much more limited in what can be done for
+// ASGs. As such, we instead build a map of tags to look for as we iterate over all ASGs in getOwnedAutoScalingGroups
+func parseAutoscaleFilterTags(clusterId string) map[string][]string {
+	if filter, ok := os.LookupEnv(customTagFilterEnvVarName); ok {
```

Comment:
What if we move the ENV setting to the main package and use this function as method (receiver) of `adapter`?
Then you do not need to use set/unset ENV, which might be error prone in concurrent tests.

Reply:
Yes, I agree we should parse the filter from Env in `controller.go` and just pass the resulting value directly to the adapter.

Reply:
So in the current state, the parsing of the ENV is closely aligned to what is being done for the instance filters. (setup in buildManifest). I had a look at moving the substance of this PR to controller.go and found a lot of dependencies that would have to change as well (such as having access to the clusterID needed for the parser).

What *if*... in controller.go we took the ENV value and passed it verbatim to a new field in the adapter. Anything previously referencing os.Lookupenv would instead be changed to look at that field?

Reply:
exactly!
Store the ENV in controller.go, pass it to the adapter and store it in the adapter.
Then `func parseAutoscaleFilterTags` -> `func (a *adapter) parseAutoscaleFilterTags` and you have access to the value from ENV in `a.customTagFilter`

Reply:
Okay, sounds good, I'll work something up shortly.

Reply:
I think the latest revision should be more along what you're thinking?